java/core/src/main/java/co/worklytics/psoxy/PsoxyModule.java

```diff
@@ -244,8 +248,27 @@ ParameterSchemaUtils parameterSchemaUtils() {
     }
 
     @Provides
-    Pseudonymizer pseudonymizer(PseudonymizerImplFactory factory, ConfigService config, co.worklytics.psoxy.rules.RuleSet ruleSet) {
-        return factory.create(factory.buildOptions(config, ruleSet.getDefaultScopeIdForSource()));
+    Pseudonymizer pseudonymizer(PseudonymizerImplFactory factory, ConfigService config, RulesUtils rulesUtils, com.avaulta.gateway.rules.RuleSet ruleSet) {
```

Comment:
I just realized this is weird. Why are we providing a `Pseudonymizer` but passing a factory inside??'
why do we have this at all, shouldn't all usages of `Provider<Pseudonymizer>` be gone in favor of constructing it explicitly like we do in `CommonRequestHandler`??